arithmetization/src/main/java/net/consensys/linea/zktracer/container/stacked/StackedSet.java

```diff
@@ -56,9 +60,17 @@ public void pop() {
   }
 
   public boolean add(E e) {
-    if (!operationsCommitedToTheConflation().contains(e)) {
-      return operationsInTransaction().add(e);
+    return operationsInTransaction().add(e);
+  }
+
+  void deleteDuplicate() {
+    final List<E> toRemove =
+        new ArrayList<>(operationsInTransaction().size() * EXPECTED_PROPORTION_OF_DUPLICATE);
+    for (E e : operationsInTransaction()) {
+      if (operationsCommitedToTheConflation().contains(e)) {
+        toRemove.add(e);
+      }
     }
-    return false;
+    toRemove.forEach(operationsInTransaction()::remove);
```

Comment:
maybe a classic loop has better perf ?

Reply:
Question: why not remove the duplicate as you encounter it ? Here it  would seem that the cost is, in the worst case, of the order of the product `cs * ts + ts` where  `cs = operationsCommitedToTheConflation().size` and `ts = operationsInTransaction().size()`

Reply:
Question: why not remove the duplicate as you encounter it ? Here it  would seem that the cost is, in the worst case, of the order of the product `cs * ts + ts` where  `cs = operationsCommitedToTheConflation().size` and `ts = operationsInTransaction().size()`

Also, since I don't know how remove works, what would happen if you had e.g. `operationsInTransaction() = [x, x]` ?

Reply:
Why not do e.g.
```java
    for (E e : operationsInTransaction()) {
      if (!operationsCommitedToTheConflation().contains(e)) {
        operationsCommittedToTheConflation().add(e);
      }
    }
```

Reply:
blabla (comment for review)

Reply:
> and not remove duplicates, just add new stuff ?

Because we don't want to add them to `operationsCommitedToTheConflation` as we might pop this transaction

> Also, since I don't know how remove works, what would happen if you had e.g. operationsInTransaction() = [x, x] ?

`operationsInTransaction` can't have duplicate : it is a set

Reply:
> Because we don't want to add them to operationsCommitedToTheConflation as we might pop this transaction

But you are calling `deleteDuplicates` duing `enter`, just before you `addAll`. Am I missing something ?

Reply:
if you look at `ModuleOperationStackedSet` (which is what is used by all stateless module), we call `deleteDuplicates` at the begining of the lineCounting.
And here, we don't know yet if we'll pop or no the transaction.
![image](https://github.com/user-attachments/assets/5d19d683-fbd3-4827-a8fc-efeb103639fd)